pkg/controllers/deprovisioning/orchestration/queue.go

```diff
@@ -0,0 +1,325 @@
+			// In this case, the error is unrecoverable, so don't requeue.
+			if errors.IsNotFound(err) && q.clock.Since(cmd.TimeAdded) > time.Second*5 {
+				return false, retry.Unrecoverable(fmt.Errorf("getting machine, %w", err))
+			}
+			waitErrs[i] = fmt.Errorf("getting node claim, %w", err)
+			continue
+		}
+		// We emitted this event when Deprovisioning was blocked on launching/termination.
+		// This does not block other forms of deprovisioning, but we should still emit this.
+		q.recorder.Publish(deprovisioningevents.Launching(nodeClaim, cmd.Reason))
+		if !nodeClaim.StatusConditions().GetCondition(v1beta1.Initialized).IsTrue() {
+			q.recorder.Publish(deprovisioningevents.WaitingOnReadiness(nodeClaim))
+			waitErrs[i] = fmt.Errorf("getting node claim, %w", err)
+			continue
+		}
+		key.Initialized = true
+		// This should only be reached once since initialization is checked at the beginning.
+		deprovisioningReplacementNodeInitializedHistogram.Observe(q.clock.Since(cmd.TimeAdded).Seconds())
+	}
+	// If we have any errors, don't continue
+	if err := multierr.Combine(waitErrs...); err != nil {
+		return true, err
+	}
+
+	// Reaching here means we know that all replacements have been provisioned.
+	// All we need to do now is get a successful delete call for each node claim,
+	// then the termination controller will handle the eventual deletion of the nodes.
+	errs := make([]error, len(cmd.Candidates))
+	workqueue.ParallelizeUntil(ctx, len(cmd.Candidates), len(cmd.Candidates), func(i int) {
+		q.recorder.Publish(deprovisioningevents.Terminating(cmd.Candidates[i].Node, cmd.Candidates[i].NodeClaim, cmd.Reason)...)
+		errs[i] = retry.Do(func() error {
+			if err := nodeclaim.Delete(ctx, q.kubeClient, cmd.Candidates[i].NodeClaim); err != nil {
+				if !errors.IsNotFound(err) {
+					return err
+				}
+			}
+			return nil
+		}, waitRetryOptions(ctx)...)
+		nodeclaim.TerminatedCounter(cmd.Candidates[i].NodeClaim, cmd.Reason).Inc()
+	})
+	// If there were any deletion failures, we should requeue.
+	// In the case where we requeue, but the timeout for the command is reached, we'll
+	return false, fmt.Errorf("terminating candidates, %w", multierr.Combine(errs...))
+}
+
+func (q *Queue) setNodesUnschedulable(ctx context.Context, isUnschedulable bool, candidates ...string) error {
+	var multiErr error
+	for _, cn := range candidates {
+		node := &v1.Node{}
+		if err := q.kubeClient.Get(ctx, client.ObjectKey{Name: cn}, node); err != nil {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("getting node, %w", err))
+		}
+
+		// node is being deleted already, so no need to un-cordon
+		if !isUnschedulable && !node.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		// already matches the state we want to be in
+		if node.Spec.Unschedulable == isUnschedulable {
+			continue
+		}
+
+		stored := node.DeepCopy()
+		node.Spec.Unschedulable = isUnschedulable
+		if err := q.kubeClient.Patch(ctx, node, client.MergeFrom(stored)); err != nil {
```

Comment:
Is it easier to just set the value as we do in other reconcilers and do an `euqality.Semantic.DeepEqual` here to reduce the amount of checks we do here.

Reply:
Just thinking about it, we don't need to a deep equal, really, since we only care about the one boolean field. Since this is how it was previously, I'm inclined to just keep it as is, since we're just not concerned with the rest of the node, just the one field.